core/kernel/ldelf_loader.c

```diff
@@ -356,26 +367,37 @@ TEE_Result ldelf_dlopen(struct user_mode_ctx *uctx, TEE_UUID *uuid,
 	uint32_t panic_code = 0;
 	uint32_t panicked = 0;
 	struct ts_session *sess = NULL;
+	struct dl_entry_arg dl_entry_arg = { };
+	void *uuid_kbuf = NULL;
 
 	assert(uuid);
 
 	usr_stack -= ROUNDUP(sizeof(*arg), STACK_ALIGNMENT);
 	arg = (struct dl_entry_arg *)usr_stack;
 
+	res = memdup_user(uuid, sizeof(*uuid), (void **)&uuid_kbuf);
+	if (res)
+		return res;
+
 	res = vm_check_access_rights(uctx,
 				     TEE_MEMORY_ACCESS_READ |
 				     TEE_MEMORY_ACCESS_WRITE |
 				     TEE_MEMORY_ACCESS_ANY_OWNER,
 				     (uaddr_t)arg, sizeof(*arg));
 	if (res) {
 		EMSG("ldelf stack is inaccessible!");
-		return res;
+		goto out;
 	}
 
-	memset(arg, 0, sizeof(*arg));
```

Comment:
First, writing constant (i.e., `LDELF_DL_ENTRY_DLOPEN`) to user memory will likely to be ugly, requiring intermediate variable. Also, by calling `clear_user()` and `PUT_USER()` multiple times, it would end up calling `check_access()` multiple times, which i think is suboptimal